src/java.base/share/classes/jdk/internal/classfile/impl/verifier/ParserVerifier.java

```diff
@@ -0,0 +1,373 @@
+                }
+                verifyAttribute(ae, a, errors);
+            }
+        }
+        switch (cfe) {
+            case CompoundElement<?> comp -> {
+                for (var e : comp) verifyAttributes(e, errors);
+            }
+            case RecordAttribute ra -> {
+                for(var rc : ra.components()) verifyAttributes(rc, errors);
+            }
+            default -> {}
+        }
+    }
+
+    private void verifyAttribute(AttributedElement ae, Attribute<?> a, List<VerifyError> errors) {
+        int payLoad = ((BoundAttribute)a).payloadLen();
+        if (payLoad != switch (a) {
+            case AnnotationDefaultAttribute aa -> valueSize(aa.defaultValue());
+            case BootstrapMethodsAttribute bma -> {
+                int l = 2;
+                for (var bm : bma.bootstrapMethods()) {
+                    l += 4 + 2 * bm.arguments().size();
+                }
+                yield l;
+            }
+            case CharacterRangeTableAttribute cra -> 2 + 14 * cra.characterRangeTable().size();
+            case CodeAttribute ca -> {
+                MethodModel mm = (MethodModel)ae;
+                if (mm.flags().has(AccessFlag.NATIVE) || mm.flags().has(AccessFlag.ABSTRACT)) {
+                    errors.add(new VerifyError("Code attribute in native or abstract %s".formatted(toString(ae))));
+                }
+                if (ca.maxLocals() < Util.maxLocals(mm.flags().flagsMask(), mm.methodTypeSymbol())) {
+                    errors.add(new VerifyError("Arguments can't fit into locals in %s".formatted(toString(ae))));
+                }
+                int l = 12 + ca.codeLength() + 8 * ca.exceptionHandlers().size();
+                for (var caa : ca.attributes()) {
+                    l += 6 + ((BoundAttribute)caa).payloadLen();
+                }
+                yield l;
+            }
+            case CompilationIDAttribute cida -> 2;
+            case ConstantValueAttribute cva -> {
+                ClassDesc type = ((FieldModel)ae).fieldTypeSymbol();
+                ConstantValueEntry cve = cva.constant();
+                if (!switch (TypeKind.from(type)) {
+                    case BooleanType, ByteType, CharType, IntType, ShortType -> cve instanceof IntegerEntry;
+                    case DoubleType -> cve instanceof DoubleEntry;
+                    case FloatType -> cve instanceof FloatEntry;
+                    case LongType -> cve instanceof LongEntry;
+                    case ReferenceType -> type.equals(ConstantDescs.CD_String) && cve instanceof StringEntry;
+                    case VoidType -> false;
+                }) {
+                    errors.add(new VerifyError("Bad constant value type in %s".formatted(toString(ae))));
+                }
+                yield 2;
+            }
+            case DeprecatedAttribute da -> 0;
+            case EnclosingMethodAttribute ema -> 4;
```

Comment:
See 4.7.7:
```
Otherwise, the value of the method_index item must be a valid index into the constant_pool table. The constant_pool entry at that index must be a CONSTANT_NameAndType_info structure ([§4.4.6](https://docs.oracle.com/javase/specs/jvms/se22/html/jvms-4.html#jvms-4.4.6)) representing the name and type of a method in the class referenced by the class_index attribute above.
```

Reply:
Yes, I'll inflate CP entries referenced from the attributes, that will do the verification.
Thanks.